### DISCUSSION TITLE
Nightly auto build and release

### DISCUSSION THREAD

**Discussion by ManKiam** [microsoft/win32metadata#1532] [General] [answered]
As a community-projection user of win32metada, I need to access latest metadata after bugfixes to continue the project.
After one week of a bugfix, I'm still waiting for win32metadata release to resume my project.

It would be great to at least support publish nightly releases in GH repo.

**Comment by mikebattista** [comment]
What projection are you building?

It's pretty easy to build a winmd yourself whenever you want if you can't wait for the next release.

**Reply by ManKiam** [reply]
[win32more](https://github.com/ynkdir/py-win32more) that uses json file generated from https://github.com/ynkdir/winmd-printer

**Reply by mikebattista** [reply]
Cool. I don't see a reason to set up nightly builds when we release pretty frequently already and you can build the winmd yourself when you need an update sooner.

**Reply by ManKiam** [reply]
I'm not familiar with .net so much. build commands and pre-requires are not obvious in README file. how can I build from source?

Should I start building using [DoAll.ps1](https://github.com/microsoft/win32metadata/blob/main/DoAll.ps1)?

**Reply by mikebattista** [reply, marked as answer]
See https://github.com/microsoft/win32metadata/blob/main/CONTRIBUTING.md#set-up-your-development-environment to set up your environment and run a full build.

**Reply by mikebattista** [reply]
I'm going to push an update soon in any case, but I'd recommend learning how to build the metadata yourself which would give you the most control and agility to make progress in your projection without blocking on us.

**Reply by ManKiam** [reply]
![image](https://user-images.githubusercontent.com/61895156/231288706-a9d3f58a-1c9e-41c1-99d3-f2966358b359.png)

**Reply by mikebattista** [reply]
@aarnott any idea what might cause this? I can't repro.

**Reply by mikebattista** [reply]
Below is ultimately what the script executes. Can you run that (or similar for your VS install) successfully?

`& 'C:\Program Files (x86)\Microsoft Visual Studio\Installer\vs_installer.exe' modify --installPath 'C:\Program Files\Microsoft Visual Studio\2022\Community' --add Microsoft.NetCore.Component.Runtime.6.0 --add Microsoft.NetCore.Component.SDK --add Microsoft.VisualStudio.Component.VC.Tools.ARM64 --add Microsoft.VisualStudio.Component.VC.Tools.x86.x64 --add Microsoft.VisualStudio.Component.Windows11SDK.22000`

**Reply by ManKiam** [reply]
![image](https://user-images.githubusercontent.com/61895156/231295142-79c1aedc-235d-4d4c-ad09-fd606c0a6838.png)

**Reply by mikebattista** [reply]
Can you run `scripts\Get-VSPath.ps1`?

**Reply by ManKiam** [reply]
![image](https://user-images.githubusercontent.com/61895156/231297074-e667453d-739f-425c-9d9f-05563ee8061a.png)

**Reply by mikebattista** [reply]
Ok. I can repro your issue now. I think the fact that you have the BuildTools edition of VS installed is confusing the script.

Below is a simplified version of the script. Could you replace the entire contents of Install-VS.ps1 with the below and try again?

```pwsh
$installerArgs = @()
$vsInstallerPath = "$env:TEMP\vs_enterprise.exe"
(New-Object System.Net.WebClient).DownloadFile("https://aka.ms/vs/17/release/vs_enterprise.exe", $vsInstallerPath)
$workloadsRequired = & "$PSScriptRoot\Get-RequiredWorkloads.ps1"
$installerArgs += $workloadsRequired | % { '--add', $_ }
& $vsInstallerPath $installerArgs
```

**Reply by ManKiam** [reply]
thats true,
isn't it possible to using buildtools (that I already have installed few required components in it) without download and install VS enterprise?

**Reply by ManKiam** [reply]
using `"C:\...\vs_installer.exe" modify --installPath "C:\...\BuildTools" --add ...` installer loaded without problem. is it necessary to install vs enterprise?

**Reply by mikebattista** [reply]
If you can install the necessary components in the BuildTools edition, I imagine that should work, though I haven't tried it.

Do all the components show up as selected in the installer window?

**Reply by ManKiam** [reply]
Yes, all components showed up and selected for buildtools.

**Reply by mikebattista** [reply]
Give a full build a try once you install the components. Some of the scripts we had issues with earlier might still have issues. I'll have to fix those if so.

**Reply by ManKiam** [reply]
this line commented:
https://github.com/microsoft/win32metadata/blob/92565ba4e528fb1ea3b90d86007a974a51026efd/DoAll.ps1#L21

and here is the result with buildtools:
![image](https://user-images.githubusercontent.com/61895156/231318439-2f9dc479-dc5b-4842-b210-c119ec8cdc01.png)
it took a long. is it downloading something?

**Reply by ManKiam** [reply]
Final result:

![image](https://user-images.githubusercontent.com/61895156/231320015-ec96316b-2b3b-49d7-ba31-f0a969480e9a.png)

**Reply by mikebattista** [reply]
nbgv was installed in your earlier screenshot. Your path might not have been updated. I would try another terminal window or reboot and try again.

**Reply by ManKiam** [reply]
after reboot and after 15 minutes of building:
```pwsh
Wmp:common - Scanning to C:\Users\Desktop\win32metadata-main\generation\WinSDK\obj\generated\common\Wmp.cs...
  Scraping constants and enums...
  Writing winmd...
C:\Users\Desktop\win32metadata-main\sources\GeneratorSdk\sdk\sdk.targets(270,5): error MSB4044: The "EmitWinmd"
task was not given a value for the required parameter "WinmdVersion". [C:\Users\Desktop\win32metadata-main\gener
ation\WinSDK\Windows.Win32.proj]

Build FAILED.

C:\Users\Desktop\win32metadata-main\sources\GeneratorSdk\sdk\sdk.targets(270,5): error MSB4044: The "EmitWinmd"
task was not given a value for the required parameter "WinmdVersion". [C:\Users\Desktop\win32metadata-main\gener
ation\WinSDK\Windows.Win32.proj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:14:54.65
Exception: C:\Users\Desktop\win32metadata-main\scripts\CommonUtils.ps1:27
Line |
  27 |          throw "Call to process exited with error"
     |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Call to process exited with error
PS C:\Users\Desktop\win32metadata-main>  
```

**Reply by mikebattista** [reply]
WinmdVersion is set below:

https://github.com/microsoft/win32metadata/blob/92565ba4e528fb1ea3b90d86007a974a51026efd/scripts/BuildMetadataBin.ps1#L28

You might still have `nbgv` issues. If you run `nbgv get-version -v AssemblyVersion` from within the repo, what does that return?
